src/client/reducers/streams.ts

```diff
@@ -307,19 +320,76 @@ function recordAdditionalStream(
 }
 
 function initializeMediaRecorder(
-  streamRecordUrl: string, stream: MediaStream,
+  roomID: string, userID: string, stream: MediaStream,
 ): StreamIdRecorder {
-  const ws = new SocketClient<RecordingSocket>(streamRecordUrl)
+
+  const peer = new Peer({
+    initiator: true,
+    config: {
+      iceServers,
+      enableInsertableStreams: true,
+      // legacy flags for insertable streams
+      forceEncodedVideoInsertableStreams: true,
+      forceEncodedAudioInsertableStreams: true,
+    },
+    channelName: PEER_DATA_CHANNEL_NAME,
+    // trickle: false,
+    // Allow the peer to receive video, even if it's not sending stream:
+    // https://github.com/feross/simple-peer/issues/95
+    offerConstraints: {
+      offerToReceiveAudio: true,
+      offerToReceiveVideo: true,
+    },
+    stream,
+  })
+
+
+  peer.once(PEER_EVENT_ERROR, (err: Error) => {
+    debug("peer record error", err)
+  })
+  peer.once(PEER_EVENT_CONNECT, () => {
+    debug("peer record connect")
+  })
+  peer.once(PEER_EVENT_CLOSE, () => {
+    debug("peer record CLOSE")
+  })
+  peer.once(PEER_EVENT_SIGNAL, (data) => {
+    if (data != null) {
+      fetch('/api/sessions/' + roomID + '/join/' + userID, {
+        method: 'POST',
+        body: btoa(JSON.stringify(data))
+      }).then((resp) => {
+        resp.json().then(data => {
+          try {
+            peer.signal(data)
+          } catch (e) {
+            debug(e)
+          }
+        })
+      });
+    }
+    debug("peer record signal", data)
+  })
+  peer.on(PEER_EVENT_TRACK, (track, stream) => {
+    debug("peer record track", track, stream)
+  })
+  peer.on(PEER_EVENT_DATA, (data) => {
+    debug("peer record data", data)
+  })
+
+
+
+  // const ws = new SocketClient<RecordingSocket>(streamRecordUrl)
   const mediaRecorder = new MediaRecorder(stream, recorderOptions)
   mediaRecorder.start(1000)
   mediaRecorder.ondataavailable = (event) => {
     if (event.data && event.data.size > 0) {
```

Comment:
Do you need MediaRecorder, because as I see, you have an instance of MediaRecorder, but on any data you skip it after commenting [this line](https://github.com/ting-tube/meeting/pull/13/files#diff-e2475458371d019bf02eb688319dcf8eR387).

If you don't need it - simply delete